Remotion/Web/Core/UI/Controls/LazyEvaluatedValidationMessageControl.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Web.UI;
+using System.Web.UI.HtmlControls;
+using Remotion.Utilities;
+
+namespace Remotion.Web.UI.Controls
+{
+  public class LazyEvaluatedValidationMessageControl : HtmlContainerControl
+  {
+    private readonly IValidator _validator;
+
+    /// <summary>Initializes a new instance of the <see cref="LazyEvaluatedValidationMessageControl" /> class using the specified tag name and validator.</summary>
+    /// <param name="tag">A string that specifies the tag name of the control. </param>
+    /// <param name="validator">The validator for which the error message should be displayed.</param>
+    public LazyEvaluatedValidationMessageControl (string tag, IValidator validator)
```

Comment:
I think it's better / more correct to require  IValidatorWithDynamicErrorMessage in the ctor and instead update ValidationError to only make a LazyControl if the IValidator is also an  IValidatorWithDynamicErrorMessage